src/main/java/ch/njol/skript/conditions/CondScriptLoaded.java

```diff
@@ -54,42 +58,42 @@ public class CondScriptLoaded extends Condition {
 	private Script currentScript;
 	
 	@Override
-	@SuppressWarnings({"unchecked"})
+	@SuppressWarnings("unchecked")
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		scripts = (Expression<String>) exprs[0];
-		currentScript = getParser().getCurrentScript();
-		if (scripts == null) {
-			Skript.error("The condition 'script loaded' requires a script name argument when used outside of script files");
-			return false;
+
+		ParserInstance parser = getParser();
+		if (parser.isActive()) {
+			if (scripts == null) // no scripts provided means use current script
+				currentScript = parser.getCurrentScript();
+		} else {
+			if (scripts == null) { // parser is inactive but no scripts were provided
+				Skript.error("The condition 'script loaded' requires a script name argument when used outside of script files");
+				return false;
+			}
 		}
```

Comment:
```suggestion
		if (scripts == null) {
			if (parser.isActive()) { // no scripts provided means use current script
				currentScript = parser.getCurrentScript();
			} else { // parser is inactive but no scripts were provided
				Skript.error("The condition 'script loaded' requires a script name argument when used outside of script files");
				return false;
			}
		}
```
1 less condition check